packages/gatsby/src/schema/graphql-engine/print-plugins.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint @typescript-eslint/no-unused-vars: ["error", { "ignoreRestSiblings": true }] */
+import * as fs from "fs-extra"
+import * as path from "path"
+import * as _ from "lodash"
+import { slash } from "gatsby-core-utils"
+import { store } from "../../redux"
+import { IGatsbyState } from "../../redux/types"
+import { requireGatsbyPlugin } from "../../utils/require-gatsby-plugin"
+
+const schemaCustomizationAPIs = new Set([
+  `setFieldsOnGraphQLNodeType`,
+  `createSchemaCustomization`,
+  `createResolvers`,
+])
+
+const excludePlugins = new Set([`internal-data-bridge`, `default-site-plugin`])
+const includePlugins = new Set([`gatsby-plugin-sharp`])
+
+// Emit file that imports required node APIs
+const schemaCustomizationPluginsPath =
+  process.cwd() + `/.cache/query-engine-plugins.js`
+
+export async function printQueryEnginePlugins(): Promise<void> {
```

Comment:
Print confuses me, we aren't we calling it `writeQueryEnginePluginsToDisk` or `storeQueryEnginePlugins`

Reply:
Essentially we generate a js file with all plugin imports and exports printed. Naming is for consistency with schema printing which does a similar thing but for GraphQL schema.

Reply:
👌